src/services/services.ts

```diff
@@ -1246,7 +1268,7 @@ namespace ts {
 
             // hostCache is captured in the closure for 'getOrCreateSourceFile' but it should not be used past this point.
             // It needs to be cleared to allow all collected snapshots to be released
-            hostCache = undefined!;
+            hostCache = hostCache.thinCopy(); // The thin copy does not cache any snapshots, just their paths, so is relatively safe to hold onto longer-term

```

Comment:
I dont think we should do this since in most cases we wont need it? (eg --declaration is not enabled this is not needed.. Its not needed if there is no need to create refactoring)

Reply:
I'm fine with removing it if we're fine with not having the cache available at that point; we can always come back and add it back if we discover perf is an issue here anyway.